lib/gen_socket_client.ex

```diff
@@ -182,7 +182,7 @@ defmodule Phoenix.Channels.GenSocketClient do
           {:ok, encoded} ->
             transport.transport_mod.push(transport.transport_pid, encoded)
             {:ok, ref}
-          error -> error
+          _error -> {:error, :encoding_error}
```

Comment:
I think this might be better because you might want to match on such an error specifically when handling (in fact we most likely do). The original error might be something arbitrary.